File: 23_C#Basics/_02_ConditionalStatementsMoreExercises/_03_Harvest/_03_Harvest.cs
using System;

public class _03_Harvest
{
    public static void Main(string[] args)
    {
        // Read the input values
        double yardSquareMeters = double.Parse(Console.ReadLine());
        double grapesPerSquareMeter = double.Parse(Console.ReadLine());
        double requiredLitersOfWine = double.Parse(Console.ReadLine());
        double numberOfWorkers = double.Parse(Console.ReadLine());

        // Calculate grape and wine production
        double grapeForWine = (yardSquareMeters * grapesPerSquareMeter) * 0.4;
        double litersOfWine = grapeForWine / 2.5;

        double wine = requiredLitersOfWine - litersOfWine;

        if (wine <= 0)
        {
            // Enough wine for everyone
            Console.WriteLine($"Good harvest this year! Total wine: {Math.Floor(litersOfWine):F0} liters.");
            Console.WriteLine($"{Math.Ceiling(Math.Abs(wine)):F0} liters left -> {Math.Ceiling(Math.Abs(wine) / numberOfWorkers):F0} liters per person.");
        }
        else
        {
            // Not enough wine
            Console.WriteLine($"It will be a tough winter! More {Math.Floor(Math.Abs(wine)):F0} liters wine needed.");
        }
    }
}

File: 01_ProgrammingBasics/_02_ConditionalStatementsMoreExercises/_03_Harvest.java
package _01_ProgrammingBasics._02_ConditionalStatementsMoreExercises;

import java.util.Scanner;

public class _03_Harvest {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read the input values
        double yardSquareMeters = Double.parseDouble(scan.nextLine());
        double grapesPerSquareMeter = Double.parseDouble(scan.nextLine());
        double requiredLitersOfWine = Double.parseDouble(scan.nextLine());
        double numberOfWorkers = Double.parseDouble(scan.nextLine());

        // Calculate grape and wine production
        double grapeForWine = (yardSquareMeters * grapesPerSquareMeter) * 0.4;
        double litersOfWine = grapeForWine / 2.5;

        double wine = requiredLitersOfWine - litersOfWine;

        // Print the result
        if (wine <= 0) {
            // Enough wine for everyone
            System.out.printf("Good harvest this year! Total wine: %.0f liters.%n", Math.floor(litersOfWine));
            System.out.printf("%.0f liters left -> %.0f liters per person.", Math.ceil(Math.abs(wine)), Math.ceil(Math.abs(wine) / numberOfWorkers));
        } else {
            // Not enough wine
            System.out.printf("It will be a tough winter! More %.0f liters wine needed.", Math.floor(Math.abs(wine)));
        }

        scan.close();
    }
}
